### ISSUE TITLE
Regression in JSON parsing in 1.7.5 compared to 1.7.4 

### ISSUE THREAD

**Issue by patrick-stephens** [fluent/fluent-bit#3511]
## Bug Report

**Describe the bug**
I have a large single-line JSON file that is no longer being parsed correctly in 1.7.5 but was in 1.7.4. Other JSON files with smaller lines parse fine with the exact same configuration.

This was found during my regression tests for Couchbase but I'm reproducing here with direct use of the official image.

The problem file is this one: https://github.com/couchbase/couchbase-fluent-bit/blob/main/test/logs/rebalance/rebalance_report_2021-03-09T20:24:32Z.json
Make sure to add a new line to it (my watcher does this automatically). I've included a processed copy in this archive with the config as well.
[Archive.zip](https://github.com/fluent/fluent-bit/files/6508892/Archive.zip)

The other smaller files in that same directory parse fine.

My parsers are here: https://github.com/couchbase/couchbase-fluent-bit/blob/main/conf/parsers-couchbase.conf#L9
```
[PARSER]
    Name         couchbase_rebalance_report
    Format       json
    Time_Key     timestamp
    Time_Format  %Y-%m-%dT%H:%M:%SZ
    Time_Keep    On 
```

**To Reproduce**

Run using `expect` to trigger failure if not parsed by detecting the `log` field.
```
[SERVICE]
    flush           1
    log_level       Info
    parsers_file    /fluent-bit/etc/parsers-couchbase.conf

[INPUT]
    Name tail
    Alias rebalance_process_tail
    Path /logs/rebalance*.json
    # These files can be very large for a single line
    Buffer_Max_Size 10mb
    # We want to just skip if we overflow rather than anything else
    Refresh_Interval 10
    Skip_Long_Lines On
    Path_Key filename
    Parser couchbase_rebalance_report
    Tag couchbase.log.rebalance
    Read_from_head On

[FILTER]
    name       expect
    match      couchbase.log.*
    # make sure we don't use the unstructured fall back of 'log'
    key_not_exists log
    key_exists filename
    action     exit

[OUTPUT]
    name  null
    match *
```

- Steps to reproduce the problem:
```
docker run --rm -it -v $PWD/test.conf:/fluent-bit/etc/fluent-bit.conf -v $PWD/rebalance_large.json:/logs/rebalance_large.json -v $PWD/conf/parsers-couchbase.conf:/fluent-bit/etc/parsers-couchbase.conf fluent/fluent-bit:1.7.5
luent Bit v1.7.5
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/05/19 12:50:12] [ info] [engine] started (pid=1)
[2021/05/19 12:50:12] [ info] [storage] version=1.1.1, initializing...
[2021/05/19 12:50:12] [ info] [storage] in-memory
[2021/05/19 12:50:12] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2021/05/19 12:50:12] [ info] [sp] stream processor started
[2021/05/19 12:50:12] [error] [filter:expect:expect.0] exception on rule #0 'key_not_exists', key 'log' exists. Record content:
{"filename":"/logs/rebalance_large.json","log":"\",\"timeTaken\":234},\"backfill\":{\"startTime\":\"2021-03-09T20:24:20.235Z\",\"completedTime\":\"2021-03-09T20:24:20.251Z\",\"timeTaken\":16},\"takeover\":{\"startTime\":false},\"persistence\":{\"startTime\":false},\"replicationInfo\":{\"ns_1@cb-example-0002.cb-example.default.svc\":{\"node\":\"ns_1@cb-example-0002.cb-example.default.svc\",\"inDocsTotal\":15,\"inDocsLeft\":0}}},{\"id\":546,\"beforeChain\":[\"ns_1@cb-example-0001.cb-example.default.svc\",\"undefined\"],\"afterChain\":[\"ns_1@cb-example-0001.cb-example.default.svc\",\"ns_1@cb-example-0002.cb-example.default.svc\"],\"move\":{\"startTime\":\"2021-03-09T20:24:20.813Z\",\"completedTime\":\"2021-03-09T20:24:21.185Z\",\"timeTaken\":372},\"backfill\":{\"startTime\":\"2021-03-09T20:24:20.952Z\",\"completedTime\":\"2021-03-09T20:24:21.027Z\",\"timeTaken\":75},\"takeover\":{\"startTime\":false},\"persistence\":{\"startTime\":false},\"replicationInfo\":{\"ns_1@cb-example-0002.cb-example.default.svc\":{\"node\":\"ns_1@cb-example-0002.cb-example.default.svc\",\"inDocsTotal\":17,\"inDocsLeft\":0}}},{\"id\":674,\"beforeChain\":[\"ns_1@cb-example-0001.cb-example.default.svc\",\"undefined\"],\"afterChain\":[\"ns_1@cb-example-0001.cb-example.default.svc\",\"ns_1@cb-example-0002.cb-example.default.svc\"],\"move\":{\"startTime\":\"2021-03-09T20:23:58.578Z\",\"completedTime\":\"2021-03-09T20:23:59.116Z\",\"timeTaken\":538},\"backfill\":{\"startTime\":\"2021-03-09T20:23:58.628Z\",\"completedTime\":\"2021-03-09T20:23:58.702Z\",\"timeTaken\":74},\"takeover\":{\"startTime\":false},\"persistence\":{\"startTime\":false},\"replicationInfo\":{\"ns_1@cb-example-0002.cb-example.default.svc\":{\"node\":\"ns_1@cb-example-0002.cb-example.default.svc\",\"inDocsTotal\":3,\"inDocsLeft\":0}}},{\"id\":857,\"beforeChain\":[\"ns_1@cb-example-0001.cb-example.default.svc\",\"undefined\"],\"afterChain\":[\"ns_1@cb-example-0002.cb-example.default.svc\",\"ns_1@cb-example-0001.cb-example.default.svc\"],\"move\":{\"startTime\":\"2021-03-09T20:24:09.956Z\",\"completedTime\":\"2021-03-09T20:24:10.768Z\",\"timeTaken\":812},\"backfill\":{\"startTime\":\"2021-03-09T20:24:10.119Z\",\"completedTime\":\"2021-03-09T20:24:10.225Z\",\"timeTaken\":106},\"takeover\":{\"startTime\":\"2021-03-09T20:24:10.364Z\",\"completedTime\":\"2021-03-09T20:24:10.403Z\",\"timeTaken\":38},\"persistence\":{\"startTime\":false},\"replicationInfo\":{\"ns_1@cb-example-0002.cb-example.default.svc\":{\"node\":\"ns_1@cb-example-0002.cb-example.default.svc\",\"inDocsTotal\":6,\"inDocsLeft\":0}}},{\"id\":985,\"beforeChain\":[\"ns_1@cb-example-0001.cb-example.default.svc\",\"undefined\"],\"afterChain\":[\"ns_1@cb-example-0002.cb-example.default.svc\",\"ns_1@cb-example-0001.cb-example.default.svc\"],\"move\":{\"startTime\":\"2021-03-09T20:23:51.695Z\",\"completedTime\":\"2021-03-09T20:23:52.479Z\",\"timeTaken\":784},\"backfill\":{\"startTime\":\"2021-03-09T20:23:51.791Z\",\"completedTime\":\"2021-03-09T20:23:51.810Z\",\"timeTaken\":20},\"takeover\":{\"startTime\":\"2021-03-09T20:23:51.977Z\",\"completedTime\":\"2021-03-09T20:23:52.059Z\",\"timeTaken\":81},\"persistence\":{\"startTime\":false},\"replicationInfo\":{\"ns_1@cb-example-0002.cb-example.default.svc\":{\"node\":\"ns_1@cb-example-0002.cb-example.default.svc\",\"inDocsTotal\":15,\"inDocsLeft\":0}}},{\"id\":231,\"beforeChain\":[\"ns_1@cb-example-0000.cb-example.default.svc\",\"undefined\"],\"afterChain\":[\"ns_1@cb-example-0000.cb-example.default.svc\",\"ns_1@cb-example-0002.cb-example.default.svc\"],\"move\":{\"startTime\":\"2021-03-09T20:24:15.976Z\",\"completedTime\":\"2021-03-09T20:24:16.334Z\",\"timeTaken\":357},\"backfill\":{\"startTime\":\"2021-03-09T20:24:15.976Z\",\"completedTime\":\"2021-03-09T20:24:16.094Z\",\"timeTaken\":118},\"takeover\":{\"startTime\":false},\"persistence\":{\"startTime\":false},\"replicationInfo\":{\"ns_1@cb-example-0002.cb-example.default.svc\"
[2021/05/19 12:50:12] [ info] [input] pausing rebalance_process_tail
[2021/05/19 12:50:12] [ warn] [engine] service will stop in 5 seconds
[2021/05/19 12:50:16] [ info] [engine] service stopped
```

1.7.4 is fine:
```
docker run --rm -it -v $PWD/test.conf:/fluent-bit/etc/fluent-bit.conf -v $PWD/rebalance_large.json:/logs/rebalance_large.json -v $PWD/conf/parsers-couchbase.conf:/fluent-bit/etc/parsers-couchbase.conf fluent/fluent-bit:1.7.4
Fluent Bit v1.7.4
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/05/19 12:50:20] [ info] [engine] started (pid=1)
[2021/05/19 12:50:20] [ info] [storage] version=1.1.1, initializing...
[2021/05/19 12:50:20] [ info] [storage] in-memory
[2021/05/19 12:50:20] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2021/05/19 12:50:20] [ info] [sp] stream processor started
[2021/05/19 12:50:20] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/05/19 12:50:20] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/05/19 12:50:20] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/05/19 12:50:20] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/05/19 12:50:20] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/05/19 12:50:20] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/05/19 12:50:20] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/05/19 12:50:20] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/05/19 12:50:20] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/05/19 12:50:20] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/05/19 12:50:20] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/05/19 12:50:20] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/05/19 12:50:20] [ info] [input chunk] skip ingesting data with 0 bytes
[2021/05/19 12:50:20] [ info] [input:tail:rebalance_process_tail] inotify_fs_add(): inode=13753214 watch_fd=1 name=/logs/rebalance_large.json
^C[2021/05/19 12:56:26] [engine] caught signal (SIGINT)
```

**Expected behavior**
No failure with `expect` finding the `log` field.

**Screenshots**

**Your Environment**
* Version used: container version 1.7.5
* Configuration: As above and included in attachment
* Environment name and version (e.g. Kubernetes? What version?): Docker MacOS
* Operating System and version: MacOS Big Sur
* Filters and plugins: Tail, Expect, Stdout and JSON parser

**Additional context**
I've been using this during CI to test expected output vs real output: https://github.com/couchbase/couchbase-fluent-bit/blob/main/test/run-tests.sh

**Comment by jcdauchy-moodys** [comment]
Should we worry if we have thousands of lines in "info" log like this ?

`[ info] [input chunk] skip ingesting data with 0 bytes`

It is quite polluting the logs

**Comment by patrick-stephens** [comment]
I'm wondering if this is affecting the long line: https://github.com/fluent/fluent-bit/issues/3491

**Comment by nokute78** [comment]
Note: The patch(https://github.com/fluent/fluent-bit/pull/3490)  to fix #3491 has been merged to master.

**Comment by nokute78** [comment]
v1.7.6 has been released and #3491 has been fixed.
https://fluentbit.io/announcements/v1.7.6/

**Comment by patrick-stephens** [comment]
I can confirm our tests now pass with 1.7.6 so appreciate the efforts involved!
